### PR TITLE
Spire101 workshop

### PR DESCRIPTION
**Pull request by walmav**
workload grpc client commandline
SPIRE 101 walkthrough markdown

### CODE REVIEW

File: doc/SPIRE101.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+### Requirement(s)
+
+#### GitHub Setup
+
+Clone the SPIRE github repo:
+
+    `git clone https://github.com/spiffe/spire`
+
+Note: If you don’t already have a GitHub account, please get setup[here](https://github.com).
+
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: Space between setup and [here]

File: doc/SPIRE101.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+# Overview
+
+Follow these walkthrough to set up your docker development environment and expose the Workload API. A workload client cmdline simulates a workload fetching SVIDs for its registered SPIFFE IDs.
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: This walkthrough will guide you through the steps needed to setup a running example of a SPIRE server and agent.  Interaction with the Workload API [Link] will be simulated via a command line tool.

File: doc/SPIRE101.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    `git clone https://github.com/spiffe/spire`
+
+Note: If you don’t already have a GitHub account, please get setup[here](https://github.com).
+
+#### Docker Setup
+
+Pull the latest SPIRE image from spiffe/spire-dev docker repository:
+
+    `docker pull spiffe/spire-dev`
+Note: If you don't already have Docker already installed, please follow these[installation instructions](https://docs.docker.com/engine/installation/) .
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: space between these and [install instructions]

File: doc/SPIRE101.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+
+
+# Overview
+
+Follow these walkthrough to set up your docker development environment and expose the Workload API. A workload client cmdline simulates a workload fetching SVIDs for its registered SPIFFE IDs.
+ 
+ ![cache manager flow](images/SPIRE101.png)
+
+### Requirement(s)
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: These should be `##` not `###` Make the test bigger

File: doc/SPIRE101.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ |socketPath             |  Unix Domain Socket file path through which workloads            |
+ |                       |  connects to Workload API                                        |                                          
+ |joinToken              |  Nonce generated by the SPIRE server to attest SPIRE agents      |
+
+
+### Walkthrough
+
+Setup your local Docker development environment, cd to the directory you cloned the SPIRE repo into and start the container 
+
+    cd ~/spire
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: mkdir ~/spire

File: doc/SPIRE101.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    make build
+
+The `spire-server` and `spire-agent` executables along with their respective subcommands have a `—help`  option that give more details on using the specific cli options. Try running help for register sub command:
+	
+	./cmd/spire-server/spire-server register --help
+
+Familiarize with the server config
+	
+	cat conf/server/server.conf
+
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: you should show the actual file here also.  Just copy an example in place.Have a link for the documentation also.

File: doc/SPIRE101.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+The build target builds spire-server, spire-agent and wlcli executable, it also builds the plugin binaries.
+    
+    make build
+
+The `spire-server` and `spire-agent` executables along with their respective subcommands have a `—help`  option that give more details on using the specific cli options. Try running help for register sub command:
+	
+	./cmd/spire-server/spire-server register --help
+
+Familiarize with the server config
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: don't use  Familiarize 

"To view the configuration file of the server"

File: doc/SPIRE101.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+Start the spire-server in background by running the command:
+
+    ./cmd/spire-server/spire-server run &
+
+Generate a join token and create a vanity spiffeID: spiffe://example.org/host
+	
+	./cmd/spire-server/spire-server token generate -spiffeID spiffe://example.org/host
+
+Familiarize with the agent config
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: To view the configuration file of the agent

File: doc/SPIRE101.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+	
+	cat conf/agent/agent.conf
+
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: Show an example of the file here also. Maybe have a link to the documentation for it.

File: doc/SPIRE101.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+Start the spire-agent in the background 
+
+    ./cmd/spire-agent/spire-agent run -joinToken <generated-join-token> &
+
+Register workload spiffeID with unix type and uid:
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: The next step is to register a Spiffe ID with a set of selectors. For the example we will use unix kernel selectors that will be mapped to a target SPIFFE ID.

File: doc/SPIRE101.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+   	
+   	./cmd/spire-server/spire-server register -parentID spiffe://example.org/host -spiffeID spiffe://example.org/host/workload selector unix:uid:1000
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: Is there a way to add new lines or "\" this is just one long text line

File: doc/SPIRE101.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+Run the grpc workload client as user workload with uid 1000
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: The registration API has been called and the target workload has been registered with the SPIRE Server.  We can call the workload API using a command line program to request a new SVID from the SPIRE server.

File: doc/SPIRE101.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    
+    su -c ./cmd/wlcli/wlcli fetchsvid
+
+Print out the generated SVID. Show the SAN SPIFFE ID URI
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: Show the path for the generated SVID.  You just have 0.svid 

should be /blah/0.svid/svid  (or whatever the path is )

File: doc/SPIRE101.md (hunk 11 above)
Comment (kunzimariano) [reply doc/SPIRE101.md]: Wrapping around triple ` plus a new line where desired should do the trick.

Comment (kunzimariano) [reply doc/SPIRE101.md]:
```
./cmd/spire-server/spire-server register -parentID spiffe://example.org/host 
-spiffeID spiffe://example.org/host/workload selector unix:uid:1000
```

File: doc/SPIRE101.md (hunk 5 above)
Comment (walmav) [reply doc/SPIRE101.md]: this is just an example of where the spire repo could be cloned..we dont need to create the directory

Comment (kunzimariano) [reply doc/SPIRE101.md]: /spire should exist already at this point since it's where the code was cloned.

File: doc/SPIRE101.md [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+
+12. Start the spire-agent as a background process by replacing **_<generated-join-token>_** with the generated join token in step #10 and running the following command.
+
+        ./cmd/spire-agent/spire-agent run -joinToken <generated-join-token> &
+
+13. The next step is to register a Spiffe ID with a set of selectors. For the example we will use unix kernel selectors that will be mapped to a target SPIFFE ID.
+   	
+        ./cmd/spire-server/spire-server register -parentID spiffe://example.org/host -spiffeID spiffe://example.org/host/workload -selector unix:uid:1000
+    At this point, the registration API has been called and the target workload has been registered with the SPIRE Server. We can now call the workload API using a command line program to request a new SVID from the SPIRE server.
+
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: Should request the SVID from the SPIRE agent not the server. agent implements the Workload API

File: doc/SPIRE101.md [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+    ```
+
+12. Start the spire-agent as a background process by replacing **_<generated-join-token>_** with the generated join token in step #10 and running the following command.
+
+        ./cmd/spire-agent/spire-agent run -joinToken <generated-join-token> &
+
+13. The next step is to register a Spiffe ID with a set of selectors. For the example we will use unix kernel selectors that will be mapped to a target SPIFFE ID.
+   	
+        ./cmd/spire-server/spire-server register -parentID spiffe://example.org/host -spiffeID spiffe://example.org/host/workload -selector unix:uid:1000
+    At this point, the registration API has been called and the target workload has been registered with the SPIRE Server. We can now call the workload API using a command line program to request a new SVID from the SPIRE server.
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: the command is still one long line. Is there anyway to break it up into multiple lines like 

```
command \
   -blah \
  -blah2 
```

File: doc/SPIRE101.md [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+
+14. Simulate the workload API interaction and retrieve the workload SVID bundle by running the wlcli commandline program. Run wlcli as user **_workload_** created in step #4 with uid 1000
+
```
Comment (y2bishop2y) [inline doc/SPIRE101.md]: command line is 2 words

**Pull request merged**